extension.js

```diff
@@ -36,34 +36,40 @@ exports.activate = function activate(context) {
 function createSnippet(snippetFilename, editor) {
     fs.readFile(snippetFilename, 'utf8', function(e, data) {
         console.log(data);
-        var jsonData = json.parse(data, null, false);
-        jsonData = addSelectedSnippet(jsonData, editor);
-        updateSnippetFile(snippetFilename, jsonData);
+        var jsonData = (data ? json.parse(data, null, false) : {});
+        addSelectedSnippetAsync(jsonData, editor).then(function(result) {
+            jsonData = result;
+            updateSnippetFile(snippetFilename, jsonData);
+        });
     });
 }
 
-function addSelectedSnippet(jsonData, editor) {
+function addSelectedSnippetAsync(jsonData, editor) {
```

Comment:
No need to append "Async" here because mostly everything in javascript is async. It's actually the opposite (e.g. `fs.readFileSync`). Basically there was no good reason this wasn't async before or why I didn't name it appropriately. :smiley: